Added User Authentication, Database and Files restructuring

hirux06 opened the pull request:
Fixes #14 : 
-
I have implemented the following features:
- User signup and login using JWT authentication.
- Database and files restructuring

Thanus-Kumaar commented on db/userModule.js:
```diff
@@ -0,0 +1,59 @@
+  
+  createUser: (username, email, password, res) => {
+    const query = "INSERT INTO users (username, email, password) VALUES (?, ?, ?)";
+    con.query(query, [username, email, password], (err, result) => {
```
Also, can you modify the `sqlconnect.js` to use promises, so that you can use much cleaner code using `await`

Thanus-Kumaar commented on db/userModule.js:
```diff
@@ -0,0 +1,59 @@
+
+const playerModule = {
+  
+  addPlayerToUserId: (userId, playerId, res) => {
```
You should call this function in the `addPlayer` function in `playerController.js` so that once it is updated in the players table it should be added to the many-to-many table too. Better remove this function and add it there as a transaction (to maintain integrity)

Thanus-Kumaar commented on db/userModule.js:
```diff
@@ -0,0 +1,59 @@
+  },
+
+  
+  getPlayersForUserId: (userId, res) => {
```
Same, move this function also to the `playerController.js`

hirux06 commented:
@Thanus-Kumaar  
Made following changes: 
- Updated `sqlconnect.js` to establish query on the basis of connection pool and return promises
- Changed the old format of all the existing controllers based on connection pool styled connection 
- Added transaction in `playerController.js` for maintaining data integrity

Thanus-Kumaar commented on config/sqlconnect.js:
```diff
@@ -1,25 +1,20 @@
-      console.log("Connected to SQL Database");
-    }
-  });
+  console.log("Connection pool created successfully for SQL Database");
```
comment this, otherwise it will be printed for all controllers.

Thanus-Kumaar commented on controllers/playerController.js:
```diff
@@ -1,23 +1,52 @@
 
 const addPlayer = async (req, res) => {
-  const { name, poscat, pos } = req.body;
+  const { name, poscat, pos, playerId } = req.body;
```
Why are you getting `playerId`, can you explain?

hirux06 replied:
playerId is for maintaining relationhip between players and users, to track their dependency  in the schema

Thanus-Kumaar commented on controllers/playerController.js:
```diff
@@ -1,23 +1,52 @@
+    return res.status(400).json({ Error: "Missing required parameters" });
+  }
+
+  const insertPlayerQuery = `INSERT INTO Players (name, positionCategory, position) VALUES (?, ?, ?) ON DUPLICATE KEY UPDATE id=LAST_INSERT_ID(id);`;
```
Can you explain this query to me? I did not understand this.

hirux06 replied:
Here, we are inserting a new record into the Players table with the values provided for name, positionCategory, and position, which are already defined in the schema. Ultimately, what this means is that, once we recieve, the above three values from the table, we will be adding this as a new player, at the position and having a position category

Thanus-Kumaar commented on controllers/userController.js:
```diff
@@ -0,0 +1,50 @@
+      return res.status(400).json({ error: "Email already registered" });
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 10);
```
```suggestion
```
No need to hash the password, we should get the password hashed from the frontend, so that it will not be captured in transit.

But add a comment saying the password should be send hashed

Thanus-Kumaar reviewed (changes requested):
Well done in changing the variable names to more understandable terms.

Thanus-Kumaar commented:
@hirux06 Also, do test if all the routes are working properly before pushing.

Thanus-Kumaar commented:
/bounty 150 @hirux06
